catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -123,18 +155,188 @@ public Fileset createFileset(
+
+    FilesetEntity filesetEntity =
+        new FilesetEntity.Builder()
+            .withName(ident.name())
+            .withId(stringId.id())
+            .withNamespace(ident.namespace())
+            .withComment(comment)
+            .withFilesetType(type)
+            // Store the storageLocation to the store. The "storageLocation" can be null for managed
+            // fileset, Gravitino will get the location based on the catalog/schema's location.
+            .withStorageLocation(storageLocation)
+            .withProperties(addManagedFlagToProperties(properties))
+            .withAuditInfo(
+                new AuditInfo.Builder()
+                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                    .withCreateTime(Instant.now())
+                    .build())
+            .build();
+
+    try {
+      store.put(filesetEntity, true /* overwrite */);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to create fileset " + ident.name(), ioe);
+    }
+
+    return new HadoopFileset.Builder()
+        .withName(ident.name())
+        .withComment(comment)
+        .withType(type)
+        // The returned storageLocation is a calculated one. If the fileset is managed and
+        // storageLocation is null, then the calculated location is the
+        // schema location + fileset name.
+        .withStorageLocation(filesetPath.toString())
+        .withProperties(filesetEntity.properties())
+        .withAuditInfo(filesetEntity.auditInfo())
+        .build();
   }
 
   @Override
   public Fileset alterFileset(NameIdentifier ident, FilesetChange... changes)
       throws NoSuchFilesetException, IllegalArgumentException {
-    throw new UnsupportedOperationException();
+    FilesetEntity filesetEntity;
+    try {
+      filesetEntity = store.get(ident, Entity.EntityType.FILESET, FilesetEntity.class);
+    } catch (NoSuchEntityException exception) {
+      throw new NoSuchFilesetException("Fileset " + ident.name() + " does not exist", exception);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load fileset " + ident.name(), ioe);
+    }
+
+    List<FilesetChange> renames =
+        Arrays.stream(changes)
+            .filter(c -> c instanceof FilesetChange.RenameFileset)
+            .collect(Collectors.toList());
+    Preconditions.checkArgument(
+        renames.size() <= 1,
+        "Only at most one rename operation is allowed once for fileset operations: "
+            + ident.name());
+
+    String newPath = null;
+    if (!renames.isEmpty()) {
+      FilesetChange.RenameFileset rename = (FilesetChange.RenameFileset) renames.get(0);
+      String newName = rename.getNewName();
+      newPath = renameStorageLocation(ident, filesetEntity, newName);
+    } else {
+      newPath = getFilesetPath(ident, filesetEntity).toString();
+    }
+
+    try {
+      FilesetEntity entity =
```

Comment:
You'd better rename `entity` as the Class `HadoopCatalogOperations` also has a field `entity`.